src/node_http_parser.cc

```diff
@@ -820,7 +784,13 @@ class Parser : public AsyncWrap, public StreamListener {
     current_buffer_len_ = nread;
     current_buffer_data_ = buf.base;
 
-    MakeCallback(cb.As<Function>(), 1, &ret);
+    v8::TryCatch try_catch(env()->isolate());
+    USE(cb.As<Function>()->Call(env()->context(), object(), 1, &ret));
+
+    if (try_catch.HasCaught()) {
```

Comment:
I don't think we need this. We just shouldn't throw if the previous js callback throws, but we shouldn't stop the flow when an error is thrown on the user handler.